### ISSUE TITLE
The current code cannot be reproduced.

### ISSUE THREAD

**Issue by TAICHIKF** [yang-ze-kang/NADER#3]
There are a lot of absolute paths in the repository. If it is open source, the repository code needs to be improved. The current code cannot be reproduced.

**Comment by TAICHIKF** [comment]
In nader/nader/train_utils/train_templates.py, there is:

cd /mnt/afs/yangzekang/ModelGen
python train_imagenet16_120.py --model_name {{model_name}} --tag 1 --code-dir {{code_dir}} --output {{train_log_dir}}"

However, the following critical files are actually missing from the code repository:
train_cifar10.py, train_cifar100.py, and train_imagenet16_120.py.

This is crucial for reproducing the code from the paper.

**Comment by yang-ze-kang** [comment]
> In nader/nader/train_utils/train_templates.py, there is:
> 
> cd /mnt/afs/yangzekang/ModelGen python train_imagenet16_120.py --model_name {{model_name}} --tag 1 --code-dir {{code_dir}} --output {{train_log_dir}}"
> 
> However, the following critical files are actually missing from the code repository: train_cifar10.py, train_cifar100.py, and train_imagenet16_120.py.
> 
> This is crucial for reproducing the code from the paper.

We have added these three files in the new version.
Thank you for point it out!

**Comment by yang-ze-kang** [comment]
> There are a lot of absolute paths in the repository. If it is open source, the repository code needs to be improved. The current code cannot be reproduced.

We try to use relative paths in the code. If you find any place where an absolute path is used by mistake, please point it out.

**Comment by TAICHIKF** [comment]
> > In nader/nader/train_utils/train_templates.py, there is:
> > cd /mnt/afs/yangzekang/ModelGen python train_imagenet16_120.py --model_name {{model_name}} --tag 1 --code-dir {{code_dir}} --output {{train_log_dir}}"
> > However, the following critical files are actually missing from the code repository: train_cifar10.py, train_cifar100.py, and train_imagenet16_120.py.
> > This is crucial for reproducing the code from the paper.
> 
> We have added these three files in the new version. Thank you for point it out!


> Thanks to the author for the supplemental code.  
>   
> **In `train_cifar100.py`:**  
> - **Training set:** Uses the full 50K training set, which is consistent with NB201.  
> - **Validation set:** Selects 5K validation samples from the test set using `SubsetRandomSampler(ids)` (via `get_val_dataloader`). This approach aligns with NB201’s methodology.  
> - **Test set:** However, it uses the *entire* 10K test set as the test set (via `get_test_dataloader`), **without using the "complementary 5K subset"** (the portion not used for validation). **This is inconsistent with NB201** and results in overlap between the validation and test sets, **compromising fairness and comparability**.  
>   
> **Training setup & augmentation:**  
> - Potential issue with extra augmentation: `RandomRotation(15)`. NB201 on CIFAR uses **only** `RandomCrop(32, padding=4)` + `RandomHorizontalFlip` + `Normalize`; it **does not include rotation**.  
>   
> **In `train_imagenet16_120.py`:**  
> - `get_test_dataloader` also directly uses the entire 6000-image set as the test set (**without** using the complementary `ds['xtest'][1]` subset).  
>   
> **Model selection across all three datasets:**  
> - The code saves the `best.pth` model based on test set accuracy (**"selecting the best epoch using the test set"**). **Directly using the test set to choose the best epoch may systematically inflate the reported test performance**.
